### PR TITLE
CLOUDP-268553: Add warning for docker engine/podman minimum versions

### PR DESCRIPTION
**Pull request by jeroenvervaeke**
## Proposed changes

Add warning for docker engine/podman minimum versions.

_Jira ticket:_ CLOUDP-268553

# Testing
- Tested locally
- Updated unit tests

### CODE REVIEW

File: internal/container/docker.go [hunk 1]
```diff
@@ -25,17 +25,20 @@ import (
...
 var ErrDockerNotFound = fmt.Errorf("%w: docker not found in your system, check requirements at https://dochub.mongodb.org/core/atlas-cli-deploy-local-reqs", ErrContainerEngineNotFound)
+var ErrDeterminingDockerVersion = errors.New("could not determine docker version")
 var errParseHealthCheck = errors.New("parsing image healthcheck failed")
 var errListContainer = errors.New("container listing failed")
 var errParsingContainer = errors.New("container parsing failed")
 var errDecodingJSON = errors.New("container decoding failed")
 var errParsingPorts = errors.New("parsing ports failed")
 var errConvertHostPort = errors.New("converting host port failed")
 var errConvertContainerPort = errors.New("converting container port failed")
+var minDockerVersion = semver.New(27, 0, 0, "", "") //nolint:mnd
```
Comment (fmenezes) [inline internal/container/docker.go:41]: Our promise is the last two, so I would stick 26

File: internal/podman/client.go [hunk 2]
```diff
@@ -24,11 +24,14 @@ import (
...
 
+	"github.com/Masterminds/semver/v3"
 	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/log"
 )
 
 var (
-	ErrPodmanNotFound = errors.New("podman not found in your system, check requirements at https://dochub.mongodb.org/core/atlas-cli-deploy-local-reqs")
+	ErrPodmanNotFound           = errors.New("podman not found in your system, check requirements at https://dochub.mongodb.org/core/atlas-cli-deploy-local-reqs")
+	ErrDeterminingPodmanVersion = errors.New("could not determine docker version")
+	minPodmanVersion            = semver.New(5, 2, 0, "", "") //nolint:mnd
```
Comment (fmenezes) [inline internal/podman/client.go:34]: you can check with Ciprian but I think we need to support latest two majors, which would mean v4.9 minimum

File: internal/container/docker.go (hunk 1 above)
Comment (jeroenvervaeke) [reply internal/container/docker.go:41]: `27.0+` is what the docs say. Should we update the docs too?

Comment (jeroenvervaeke) [reply internal/container/docker.go:41]: https://www.mongodb.com/docs/atlas/cli/current/atlas-cli-deploy-local/

Comment (fmenezes) [reply internal/container/docker.go:41]: yes our current epic (linux stability) will update the docs

Comment (fmenezes) [reply internal/container/docker.go:41]: dicussed offline, let's keep only latest

**Review by fmenezes — APPROVED** [review]

**Review by fmenezes — APPROVED** [review]

**Pull request merged**